Qfront/src/redux/product/filteredProducts/actions.js

```diff
@@ -0,0 +1,62 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+import { baseUrl, productsEndPointApi } from "../../api";
+
+const fetchFilteredProductsApi = `${productsEndPointApi}/by/search`;
+
+// Create action to fetch filtered products
+export const getFilteredProducts = createAsyncThunk("filteredProducts/getFilteredProducts",
+  async( { skip, limit, filters }, { rejectWithValue }) => {
+    try {
+      const data = { skip, limit, filters }
+      const response = await baseUrl.post(
+          fetchFilteredProductsApi,
+          { data },
+          {
+            headers: {
+              Accept: 'application/json',
+              'Content-type': 'application/json',
+              }
+          } )
+      return response.data;
+    } catch (error) {
+      return rejectWithValue(error)
+    }
+});
+
+// Create an action to update filtered products
+export const updateFilteredProducts = createAsyncThunk('filteredProducts/updateFilteredProducts',
+  async( { skip, limit, filters }, { rejectWithValue }) => {
+    try {
+      // This action receives data with new parameters
+      // Also returns size
+      const data = { skip, limit, filters}
+      const response = await baseUrl.post(
+          fetchFilteredProductsApi,
+          { data },
+          {
+            headers: {
+              Accept: 'application/json',
+              'Content-type': 'application/json',
+              }
+          } )
+      return response.data;
+    } catch (error) {
+      return rejectWithValue(error)
+    }
+});
+
+// create action to list products
+export const listProducts = createAsyncThunk("loadProducts/listProducts", async( { params }, { rejectWithValue } ) => {
+  try {
+    const query = {
```

Comment:
Is params always supplied to this action?
I see nothing passed [here](https://github.com/Oniocha/Qargo-Ecommerce/pull/11/files#diff-d70b1568fcc78bad0e342846f883db2a81f86af4529b3997a34968452c299d37R26)

In that case the params would be undefined and the request will be made with it. Only construct the query object if params is present.